HW_3/Task2.py

```diff
@@ -0,0 +1,19 @@
+# Задача 18: Требуется найти в массиве A самый близкий по
+# величине элемент к заданному числу X. Если таких элементов несколько,
+# вы вывести один любой.
+# Пользователь в первой строке вводит натуральное число N – количество элементов в массиве.
+# В последующих  строках записаны N целых чисел Ai (можно использовать псевдогенерацию).
+# Последняя строка содержит число X
+
+# *Пример:*
+
+# 5
+#     7 -2 3 5 1
+#     6
+#     -> 7
+
+N = int(input('Введите количество элементов в массиве N:'))
+a = [int(input('Введите число массива:')) for i in range(N)]
+x = int(input('Введите заданное число X:'))
+b = [abs(a[i]-x) for i in range(len(a))]
```

Comment:
У вас ведь уже есть параметр, который содержат длину списка
```suggestion
b = [abs(a[i] - x) for i in range(N)]
```